contracts/transfer/circuits/src/gadgets/merkle.rs

```diff
@@ -6,44 +6,48 @@
 
 use dusk_plonk::prelude::*;
 use plonk_gadgets::AllocatedScalar;
-use poseidon252::merkle_proof::{merkle_opening_gadget, PoseidonBranch};
+use poseidon252::tree::merkle_opening;
+use poseidon252::tree::PoseidonBranch;
 
 /// Prove the knowledge of the position of the note in
 /// the merkle tree.
 pub fn merkle(
     composer: &mut StandardComposer,
-    branch: PoseidonBranch,
+    branch: PoseidonBranch<31>,
```

Comment:
Depth 31 seems incorrect to me. This is a very big tree. For example; the notes tree is depth `17`